### ISSUE TITLE
Fails to compile from AUR (boost 1.69 related?)

### ISSUE THREAD

**Issue by traverseda** [rethinkdb/rethinkdb#6732]
Trying to build the AUR package from here: https://aur.archlinux.org/packages/rethinkdb/

```
src/clustering/administration/auth/user.cc: In member function ‘bool auth::user_t::has_connect_permission() const’:
src/clustering/administration/auth/user.cc:268:50: error: cannot convert ‘boost::logic::tribool’ to ‘bool’ in return
     return m_global_permissions.get_connect() || false;
                                                  ^~~~~
make[1]: *** [src/build.mk:390: build/release/obj/clustering/administration/auth/user_context.o] Error 1
src/clustering/administration/auth/scram_authenticator.cc: In static member function ‘static auth::username_t auth::scram_authenticator_t::saslname_decode(const string&)’:
src/clustering/administration/auth/scram_authenticator.cc:235:24: warning: this statement may fall through [-Wimplicit-fallthrough=]
                 offset += 2;
                 ~~~~~~~^~~~
src/clustering/administration/auth/scram_authenticator.cc:236:13: note: here
             default:
             ^~~~~~~
make[1]: *** [src/build.mk:390: build/release/obj/clustering/administration/auth/user.o] Error 1
make: *** [Makefile:53: make] Error 2

```

[log.txt](https://github.com/rethinkdb/rethinkdb/files/3065002/log.txt)

**Comment by Markus00000** [comment]
Build succeeded after downgrading to `boost-1.68.0-2` and `boost-libs-1.68.0-2`.

**Comment by jasonkhanlar** [comment]
I have been waiting for months to downgrade, uncertain what to do. I've read https://wiki.archlinux.org/index.php/Downgrading_packages

How do you downgrade to boost-1.68.0-2 and boost-libs-1.68.0-2? I no longer have those files in my local system cache as I ran out of disk space and deleted them

**Comment by srh** [comment]
I had made a change once (upon a time) to get rid of tribool, but it had a bug and failed some test, or caused a crash.  As a result I reverted the change.  I could try again and remove it.

It's only used in a few places, and (in my opinion) is an unnecessary dependency, and its API is too implicit in how it works (in C++).

**Comment by ezrast** [comment]
@jasonkhanlar or anyone else with this issue on Arch, you can fetch old Boost packages from the Arch Linux Archive like so:
```
sudo pacman -U \
  https://archive.archlinux.org/packages/b/boost/boost-1.68.0-2-x86_64.pkg.tar.xz \
  https://archive.archlinux.org/packages/b/boost-libs/boost-libs-1.68.0-2-x86_64.pkg.tar.xz
```
Running the above resolved the tribool issue for me. More info on the Arch Linux Archive is at https://wiki.archlinux.org/index.php/Arch_Linux_Archive.

**Comment by srh** [comment]
Note to self: The right fix to make here (for 2.4, immediate releases) is to just do an explicit conversion, instead of the refactoring I had done previously (which had some sort of bug).  Long-run, I want to get rid of tribool.

**Comment by srh** [comment]
This is fixed in the 2.3.x branch with #6751 .  It was already fixed in 2.4.x and next.  (The bug I mentioned had already been found and fixed.)

boost::tribool usage is removed (and replaced with enum class tribool).

**Comment by srh** [comment]
Note that I haven't actually tested this on Arch Linux -- so if there is another build error, well, there's that.  But it won't be the one you see here.
